### PR TITLE
flow: move A and B Nodes to gremlin steps

### PR DESCRIPTION
**Pull request by masco-enovance**

### CODE REVIEW

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by masco-enovance** [comment]
run skydive-selenium-tests

**Comment by masco-enovance** [comment]
run skydive-unit-tests

**Comment by masco-enovance** [comment]
run skydive-unit-tests

**Comment by masco-enovance** [comment]
run skydive-selenium-tests

**Comment by masco-enovance** [comment]
run skydive-selenium-tests

**Comment by masco-enovance** [comment]
run skydive-selenium-tests

File: flow/enhancers/graph.go [hunk 1]
```diff
@@ -60,17 +60,10 @@ func (gfe *GraphFlowEnhancer) getNodeTID(mac string) (tid string) {
...
-	if f.Link == nil {
-		return
-	}
-	if f.ANodeTID == "" {
-		f.ANodeTID = gfe.getNodeTID(f.Link.A)
-	}
-	if f.BNodeTID == "" {
-		f.BNodeTID = gfe.getNodeTID(f.Link.B)
-	}
+	//for future. currently, nothing to do.
```
[the remark this answers is not in the document]
Comment (masco-enovance) [reply flow/enhancers/graph.go]: done

File: flow/enhancers/neutron.go [hunk 2]
```diff
@@ -60,17 +60,10 @@ func (nfe *NeutronFlowEnhancer) getNodeTID(mac string) (tid string) {
...
-	if f.Link == nil {
-		return
-	}
-	if f.ANodeTID == "" {
-		f.ANodeTID = nfe.getNodeTID(f.Link.A)
-	}
-	if f.BNodeTID == "" {
-		f.BNodeTID = nfe.getNodeTID(f.Link.B)
-	}
+	//for future. currently, nothing to do.
```
[the remark this answers is not in the document]
Comment (masco-enovance) [reply flow/enhancers/neutron.go]: done

File: gremlin/traversal/flows.go [hunk 3]
```diff
@@ -138,8 +138,8 @@ func (f *FlowTraversalStep) Out(s ...interface{}) *traversal.GraphTraversalV {
 	defer f.GraphTraversal.RUnlock()
 
 	for _, flow := range f.flowset.Flows {
-		if flow.BNodeTID != "" && flow.BNodeTID != "*" {
-			m["TID"] = flow.BNodeTID
+		if flow.Link.B != "" {
+			m["MAC"] = flow.Link.B
```
[the remark this answers is not in the document]
Comment (masco-enovance) [reply gremlin/traversal/flows.go]: not getting this, could you please give more info. thanks!

[the remark this answers is not in the document]
Comment (masco-enovance) [reply gremlin/traversal/flows.go]: done!

File: statics/js/components/graph-layout.js [hunk 4]
```diff
@@ -125,7 +125,7 @@ var LinkLabelLatency = Vue.extend({
     },
 
     flowCategoryKey: function(flow) {
-      return `a=${flow.ANodeTID} b=${flow.BNodeTID} app=${flow.Application}`;
+      return `app=${flow.Application}`;
```
Comment (safchain) [inline statics/js/components/graph-layout.js]: I think here Application is not enough, You could use link.A/B or network.A/B

Comment (masco-enovance) [reply statics/js/components/graph-layout.js]: done. added link.A

**Comment by lebauce** [comment]
@masco-enovance Can you please rebase on master ?

**Comment by masco-enovance** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun skydive-functional-tests-backend-elasticsearch

**Pull request merged**